frontend-build-pipeline.yml

```diff
@@ -0,0 +1,52 @@
+trigger:
+- main
+- staging
+- development
+
+pool:
+  vmImage: ubuntu-latest
+
+steps:
+
+# Node.js is required to build Angular apps
+- task: NodeTool@0
+  inputs:
+    versionSpec: '12.14'
+  displayName: 'Install Node.js'
+
+# --dev option installs devDependencies from package.json
+# probably needed for running the Karma tests
+#
+# angular-devkit/build-angular
+# https://stackoverflow.com/questions/50333003/could-not-find-module-angular-devkit-build-angular
+- script: |
+    npm install --save-dev @angular-devkit/build-angular
+    npm install -g @angular/cli
+    npm install
+    ng build --prod
+  displayName: 'npm Install and Build'
+
+# continueOnError for now since unit tests fail
+- script: |
+    npx ng test --karma-config karma.conf.js --watch=false
+  displayName: 'Run Unit Tests'
+  continueOnError: True
```

Comment:
another option besides continueOnError is edit the test files to make all the currently failing ones skip. nearly all modern test frameworks have easy ways to mark tests for being skipped. i recommend that approach since it puts the ball more in the devs' court and doesn't affect the pipeline, but there's room for disagreement

Reply:
From the research I've done about this, it seems that these unit tests are coming from auto-generated spec.ts files when the project was created. I looked up how to skip these but it seems that the answers say to just delete the spec.ts files. Other answers mentioned `ng new --skipTests` when creating the angular project to avoid generating these files. I'm not sure if the devs would want to keep these files or not, but for the time being I'll leave  `continueOnError: True` until I know more.

Reply:
the tests are written with jasmine the jasmine way to skip tests with by [marking them as "pending"](https://jasmine.github.io/2.3/introduction#section-Pending_Specs). i suggest the `pending()` approach so you can give a reason: that the test was failing before the iteration began.

Reply:
I see, thanks for the link. I added the pending message to each unit test that was failing.